### ISSUE TITLE
Are there any plans to support WebAssembly?

### ISSUE THREAD

**Issue by baoyachi** [mmastrac/rust-ctor#297]

**Comment by mmastrac** [comment]
I'm not sure what the best approach might be for WebAssembly. Can a WebAssembly module run code at startup?

Happy to take contributions!

**Comment by mpapierski** [comment]
@mmastrac Yes, Wasm can define a `start` section that points at a function index, and it's supposed to be invoked automatically upon module instantiation as defined here: https://webassembly.github.io/spec/core/syntax/modules.html#syntax-start

From what I know there's an ongoing debate on the rust core and how to approach this best as rust is 'no life before main' and the start section violates that. If there's any way to deal with this with linker magic, that'd be great.

**Comment by mmastrac** [comment]
Fixed in #313

**Comment by baoyachi** [comment]
@mmastrac Thx.

**Comment by mpapierski** [comment]
@mmastrac I tried ctor 0.3.2 but it does not generate `(start)` section for `wasm32-unknown-unknown` target. Am I doing something wrong?

**Comment by mmastrac** [comment]
@mpapierski Odd. There's a `test-wasm` project in this repo. I tried running the following and it worked as expected:

```
wasm-pack build --target deno tests-wasm/
❯ deno eval "import './tests-wasm/pkg/tests_wasm.js';"
hello!
```

Did you get a `__wbindgen_start` symbol?
